jquery.flot.composeImages.js

```diff
@@ -0,0 +1,171 @@
+            result = -1; //nothing to do if called without sources
+        } else {
+            var minX = sources[0].genLeft;
+            var minY = sources[0].genTop;
+            var maxX = sources[0].genRight;
+            var maxY = sources[0].genBottom;
+            var i = 0;
+
+            for (i = 1; i < sources.length; i++) {
+                if (minX > sources[i].genLeft) {
+                    minX = sources[i].genLeft;
+                }
+
+                if (minY > sources[i].genTop) {
+                    minY = sources[i].genTop;
+                }
+            }
+
+            for (i = 1; i < sources.length; i++) {
+                if (maxX < sources[i].genRight) {
+                    maxX = sources[i].genRight;
+                }
+
+                if (maxY < sources[i].genBottom) {
+                    maxY = sources[i].genBottom;
+                }
+            }
+
+            if ((maxX - minX <= 0) || (maxY - minY <= 0)) {
+                result = -2;
+            } else {
+                destination.width = Math.round(maxX - minX);
+                destination.height = Math.round(maxY - minY);
+
+                for (i = 0; i < sources.length; i++) {
+                    sources[i]["xCompOffset"] = sources[i].genLeft - minX;
+                    sources[i]["yCompOffset"] = sources[i].genTop - minY;
+                }
+            }
+        }
+        return result;
+    }
+
+    function copyImgsToCanvas(sources, destination) {
+        var prepareImagesResult = prepareImagesToBeComposed(sources, destination);
+        if (prepareImagesResult === 0) {
+            var destinationCtx = destination.getContext('2d');
+
+            for (var i = 0; i < sources.length; i++) {
+                destinationCtx.drawImage(sources[i], sources[i].xCompOffset, sources[i].yCompOffset);
+            }
+        }
+        return prepareImagesResult;
+    }
+
+    function generateTempImageFromCanvasOrSvg(srcCanvasOrSvg, destImg) {
+        if (srcCanvasOrSvg.tagName === "CANVAS") {
+            copyCanvasToImg(srcCanvasOrSvg, destImg);
+        }
+
+        if (srcCanvasOrSvg.tagName === "svg") {
+            copySVGToImg(srcCanvasOrSvg, destImg);
+        }
+
+        //destImg will be used as temp image, so add to it some properties about position on page.
+        destImg["genLeft"] = srcCanvasOrSvg.getBoundingClientRect().left;
+        destImg["genTop"] = srcCanvasOrSvg.getBoundingClientRect().top;
+        destImg["genRight"] = srcCanvasOrSvg.getBoundingClientRect().right;
+        destImg["genBottom"] = srcCanvasOrSvg.getBoundingClientRect().bottom;
+    }
+
+    function getExecuteImgComposition(tempImgs, destinationCanvas) {
+        return function executeImgComposition(result) {
+            var compositionResult = copyImgsToCanvas(tempImgs, destinationCanvas);
+            return compositionResult;
+        };
+    }
+
+    function failureCallback(error) {
+        error = 0;
+        return -100; //a negative number
```

Comment:
The need of this comment can be removed by defining some constants.

Reply:
done